client/extensions/woocommerce/app/dashboard/manage-orders-view.js

```diff
@@ -5,107 +5,90 @@
 import LabelsSetupNotice from 'woocommerce/woocommerce-services/components/labels-setup-notice';
-import { fetchOrders } from 'woocommerce/state/sites/orders/actions';
-import { fetchReviews } from 'woocommerce/state/sites/reviews/actions';
-import formatCurrency from 'lib/format-currency';
-import {
-	areOrdersLoading,
-	areOrdersLoaded,
-	getNewOrdersWithoutPayPalPending,
-	getNewOrdersWithoutPayPalPendingRevenue,
-} from 'woocommerce/state/sites/orders/selectors';
+import { fetchCounts } from 'woocommerce/state/sites/data/counts/actions';
 import { getCurrentUser } from 'state/current-user/selectors';
 import { getSelectedSiteWithFallback } from 'woocommerce/state/sites/selectors';
 import { getLink } from 'woocommerce/lib/nav-utils';
-import { getPaymentCurrencySettings } from 'woocommerce/state/sites/settings/general/selectors';
-import { getTotalReviews } from 'woocommerce/state/sites/reviews/selectors';
 import InventoryWidget from './widgets/inventory-widget';
 import ShareWidget from 'woocommerce/components/share-widget';
 import QuerySettingsGeneral from 'woocommerce/components/query-settings-general';
 import StatsWidget from './widgets/stats-widget';
 
 class ManageOrdersView extends Component {
 	static propTypes = {
+		fetchCounts: PropTypes.func,
 		site: PropTypes.shape( {
 			ID: PropTypes.number.isRequired,
 			slug: PropTypes.string.isRequired,
 			URL: PropTypes.string.isRequired,
 		} ),
-		fetchOrders: PropTypes.func,
-		currency: PropTypes.shape( {
-			value: PropTypes.string,
-		} ),
-		orders: PropTypes.array,
-		ordersRevenue: PropTypes.number,
-		ordersLoading: PropTypes.bool,
-		ordersLoaded: PropTypes.bool,
+		totalNewOrders: PropTypes.number,
+		totalPendingReviews: PropTypes.number,
+		translate: PropTypes.func.isRequired,
 		user: PropTypes.shape( {
 			display_name: PropTypes.string,
 			username: PropTypes.string.isRequired,
 		} ),
 	};
 
 	componentDidMount() {
-		const { ordersLoaded, site } = this.props;
+		const { site } = this.props;
 
 		if ( site && site.ID ) {
-			this.fetchData( { siteId: site.ID, ordersLoaded } );
+			this.fetchData();
 		}
 	}
 
-	componentWillReceiveProps( newProps ) {
+	componentDidUpdate( prevProps ) {
 		const { site } = this.props;
-		const newSiteId = newProps.site ? newProps.site.ID : null;
-		const oldSiteId = site ? site.ID : null;
+		const siteId = site ? site.ID : null;
+		const oldSiteId = prevProps.site ? prevProps.site.ID : null;
 
-		if ( oldSiteId !== newSiteId ) {
-			this.fetchData( { ...newProps, siteId: newSiteId } );
+		if ( siteId && oldSiteId !== siteId ) {
+			this.fetchData();
 		}
 	}
 
-	fetchData = ( { siteId, ordersLoaded } ) => {
-		if ( ! ordersLoaded ) {
-			this.props.fetchOrders( siteId );
+	fetchData = () => {
+		const { isLoaded, site } = this.props;
+		if ( ! isLoaded ) {
+			this.props.fetchCounts( site.ID );
 		}
-		this.props.fetchReviews( siteId, { status: 'pending' } );
```

Comment:
This is pretty great stuff to see, and this counts endpoint is going to be useful in new-dash too 😄